mesonbuild/modules/python.py

```diff
@@ -334,6 +334,20 @@ def _find_installation_impl(self, state: 'ModuleState', display_name: str, name_
             if not python.found() and name_or_path in {'python2', 'python3'}:
                 python = PythonExternalProgram('python')
 
+                if python.found():
+                    # Ensure found python is the right version
+                    python.name = name_or_path
+                    if python.sanity(state):
+                        python.name = 'python'
+                        return python
+
+                    sanitymsg = f'python version {name_or_path[-1]} was not found on this system'
+                    if required:
+                        raise mesonlib.MesonException(sanitymsg)
+
+                    mlog.warning(sanitymsg, location=state.current_node)
+                    return NonExistingExternalProgram()
```

Comment:
Instead of repeating lots of logic here, I think it probably makes sense to just pass `name=name_or_path, command=python` to the external program class. This would also make us log "Program python2 found: YES (/usr/bin/python)" instead of overwriting that message with the generic name.